common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java

```diff
@@ -114,18 +114,40 @@ public void handleCraftingShaped(final PacketWrapper wrapper) throws Exception {
             wrapper.passthrough(Type.DOUBLE); // Y
             wrapper.passthrough(Type.DOUBLE); // Z
             wrapper.passthrough(Type.FLOAT); // Power
-            final int blocks = wrapper.passthrough(Type.VAR_INT);
+
+            final int blocks = wrapper.read(Type.VAR_INT);
+            byte[][] toBlow = new byte[blocks][3];
             for (int i = 0; i < blocks; i++) {
-                wrapper.passthrough(Type.BYTE); // Relative X
-                wrapper.passthrough(Type.BYTE); // Relative Y
-                wrapper.passthrough(Type.BYTE); // Relative Z
+                toBlow[i] = new byte[]{
+                    wrapper.read(Type.BYTE), // Relative X
+                    wrapper.read(Type.BYTE), // Relative Y
+                    wrapper.read(Type.BYTE) // Relative Z
+                };
+            }
+
+            float knockbackX = wrapper.read(Type.FLOAT); // Knockback X
+            float knockbackY = wrapper.read(Type.FLOAT); // Knockback Y
+            float knockbackZ = wrapper.read(Type.FLOAT); // Knockback Z
+
+            Integer blockInteraction = wrapper.read(Type.VAR_INT); // Block interaction type
+            // 0 = keep, 1 = destroy, 2 = destroy_with_decay, 3 = trigger_block
```

Comment:
What's the trigger one used for exactly? Presumably that one also wouldn't destroy blocks, though I can't check at the minute

Reply:
The trigger type is used by the breeze to toggle blocks like buttons, doors, levers... It doesn't destroy any blocks.
I have done some basic testing with this PR and everything worked as expected!